### PR TITLE
feature 37108 - Describing system processes

### PR DESCRIPTION
**Pull request by Rick-Ritense**

### CODE REVIEW

**Comment by marijnritense** [comment]
Missing the update to `SUMMARY.md`

**Comment by Rick-Ritense** [comment]
You are right ... working on it now

File: using-valtimo/process/systemprocesses.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# System processes
+
+System processes are a special kind of process. A system process describes an often used process
```
Comment (marijnritense) [inline using-valtimo/process/systemprocesses.md:3]: I would avoid fluff. Rather than stating they're a special kind of process, describe what they are right away. Maybe rephrase to:

System processes are processes that are critical to the functioning of Valtimo itself. Since these processes should be handled with care, these processes are flagged as read-only by default. When a process is read-only, it is not possible to update the process or deploy it.

File: using-valtimo/process/systemprocesses.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+System processes are a special kind of process. A system process describes an often used process 
+that is used by the system. These process should be handled with care so when a process
+is a system process it will be read-only by default. When a process is read-only it is not possible
+to update the process or deploy it.
+
+![Read only system process](img/read-only-system-process.png)
+
+A process can be marked as a system process by setting a property called `systemProcess`
+to the value `true` on the model itself under the `Extensions` tab
```
Comment (marijnritense) [inline using-valtimo/process/systemprocesses.md:11]: `the value` can be omitted here.

Comment (marijnritense) [reply using-valtimo/process/systemprocesses.md:11]: Also missing punctuation at the end of the sentence.

File: using-valtimo/process/systemprocesses.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+![Setting system process property](img/setting-system-process-property.png)
+
+## Make system process updatable
+It is possible to make a system process updatable / deployable. To override this a property has to
```
Comment (marijnritense) [inline using-valtimo/process/systemprocesses.md:16]: By default, system processes are marked as read-only. To override this, a property has to...

File: using-valtimo/process/systemprocesses.md (hunk 2 above)
Comment (Rick-Ritense) [reply using-valtimo/process/systemprocesses.md:11]: You mean that exact text only?

File: using-valtimo/process/systemprocesses.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+be set in the `application.yml` of the backend implementation. The property is called
+`valtimo.process.systemProcessUpdatable` and this should have the value `true`. The default value of
+this property is `false`
```
Comment (marijnritense) [inline using-valtimo/process/systemprocesses.md:19]: Missing punctuation

File: using-valtimo/process/systemprocesses.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+```yaml
+valtimo:
+    process:
+        systemProcessUpdatable: true
+```
+
+When a system process is updatable it is possible to make changes to the process and the 
+`Deploy` button is available again. But a message is still shown to warns the user to be
+careful when making changes to a system process
```
Comment (marijnritense) [inline using-valtimo/process/systemprocesses.md:29]: Missing punctuation.

File: using-valtimo/process/systemprocesses.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+this property is `false`
+
+```yaml
+valtimo:
+    process:
+        systemProcessUpdatable: true
+```
+
+When a system process is updatable it is possible to make changes to the process and the 
+`Deploy` button is available again. But a message is still shown to warns the user to be
```
Comment (marijnritense) [inline using-valtimo/process/systemprocesses.md:28]: Maybe rephrase this last sentence to:

A message is shown to warn the user about changing a system process.

File: using-valtimo/process/systemprocesses.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# System processes
+
+System processes are processes that are critical to the functioning of Valtimo itself. Since these 
+processes should be handled with care, these processes are flagged as read-only by default. When a 
+process is read-only, it is not possible to update the process or deploy it.
+
+![Read only system process](img/read-only-system-process.png)
+
+A process can be marked as a system process by setting a property called `systemProcess`
+to `true` on the model itself under the `Extensions` tab
```
Comment (marijnritense) [inline using-valtimo/process/systemprocesses.md:10]: Still missing a . at the end of this sentence.

File: release-notes/major9/09.21.0/valtimo-backend-libraries.md [hunk 8]
```diff
@@ -8,6 +8,12 @@ The following features were added:
 
   The `isActive` property of the `CamundaProcessJsonSchemaDocumentInstance` indicates whether the process is still running
 
+* **System processes**
+
+  Processes can now be marked as a system process. System processes are processes that are critical 
+  to the functioning of Valtimo itself. More information about this feature can be found 
+  [here](/using-valtimo/process/systemprocesses.md)
```
Comment (marijnritense) [inline release-notes/major9/09.21.0/valtimo-backend-libraries.md:15]: Missing a . at the end of the sentence.

File: release-notes/major9/09.18.0/valtimo-backend-libraries.md [hunk 9]
```diff
@@ -31,6 +31,10 @@ The following features were added:
     * This plugin can make a connection to the Zaken API. This plugin supports the action to link documents that have been uploaded 
     to the Documenten API to the Zaak of the current case
```
Comment (marijnritense) [inline release-notes/major9/09.18.0/valtimo-backend-libraries.md:32]: Missing a .

File: release-notes/major9/09.18.0/valtimo-backend-libraries.md [hunk 10]
```diff
@@ -31,6 +31,10 @@ The following features were added:
...
 
+* **Improved CamundaProcessJsonSchemaDocumentInstance**
+
+  The `isActive` property of the `CamundaProcessJsonSchemaDocumentInstance` indicates whether the process is still running
```
Comment (marijnritense) [inline release-notes/major9/09.18.0/valtimo-backend-libraries.md:36]: Missing a .

**Review by marijnritense — APPROVED** [review]
